app/services/backup_code_generator.rb

```diff
@@ -0,0 +1,70 @@
+require 'digest'
+
+class BackupCodeGenerator
+  attr_reader :length
+
+  NUMBER_OF_CODES = 10
+
+  def initialize(user, length: 2, split: 4)
+    @length = length
+    @split = split
+    @user = user
+  end
+
+  def generate
+    delete_existing_codes
+    generate_new_codes
+  end
+
+  def create
+    @user.save
+    generate
+  end
+
+  def verify(plaintext_code)
+    backup_code = normalize(plaintext_code)
+    code = BackupCodeConfiguration.find_with_code(code: backup_code, user_id: @user.id)
+    return unless code_usable?(code)
+    code.update!(used: true, used_at: Time.zone.now)
+    true
+  end
+
+  def delete_existing_codes
+    @user.backup_code_configurations.destroy_all
+  end
+
+  def generate_new_codes
+    result = []
+    (0..(NUMBER_OF_CODES - 1)).each do
+      code = backup_code
+      result.push code
+      save_code(code)
+    end
+    result
+  end
+
+  private
+
+  def code_usable?(code)
+    code && !code.used
+  end
+
+  # This method smells of :reek:FeatureEnvy
+  def save_code(code)
+    rc = BackupCodeConfiguration.new
```

Comment:
Style: or 
```ruby
def save_code(code)
  rc = BackupCodeConfiguration.new(code: code, user_id: @user.id)
  rc.save
end
```

or 
```ruby
def save_code(code)
  @user.backup_code_configurations.create!(code: code)
end
```

`used` defaults to `false`, so we don't need to explicitly set it here.

Reply:
I'll choose door number 2.  Clever!